Sources/Core/Microsoft.StreamProcessing/Aggregates/SortedMultisetAggregateBase.cs

```diff
@@ -24,21 +24,27 @@ protected SortedMultisetAggregateBase(IComparerExpression<T> comparer, QueryCont
         public Expression<Func<SortedMultiSet<T>>> InitialState() => initialState;
 
         private static readonly Expression<Func<SortedMultiSet<T>, long, T, SortedMultiSet<T>>> acc
-            = (set, timestamp, input) => set.Add(input);
+            = (set, timestamp, input) => Apply(set, s => s.Add(input));
         public Expression<Func<SortedMultiSet<T>, long, T, SortedMultiSet<T>>> Accumulate() => acc;
 
         private static readonly Expression<Func<SortedMultiSet<T>, long, T, SortedMultiSet<T>>> dec
-            = (set, timestamp, input) => set.Remove(input);
+            = (set, timestamp, input) => Apply(set, s => s.Remove(input));
         public Expression<Func<SortedMultiSet<T>, long, T, SortedMultiSet<T>>> Deaccumulate() => dec;
 
         private static readonly Expression<Func<SortedMultiSet<T>, SortedMultiSet<T>, SortedMultiSet<T>>> diff
-            = (leftSet, rightSet) => leftSet.RemoveAll(rightSet);
+            = (leftSet, rightSet) => Apply(leftSet, s => s.RemoveAll(rightSet));
         public Expression<Func<SortedMultiSet<T>, SortedMultiSet<T>, SortedMultiSet<T>>> Difference() => diff;
 
         private static readonly Expression<Func<SortedMultiSet<T>, SortedMultiSet<T>, SortedMultiSet<T>>> sum
-            = (leftSet, rightSet) => leftSet.AddAll(rightSet);
+            = (leftSet, rightSet) => Apply(leftSet, s => s.AddAll(rightSet));
         public Expression<Func<SortedMultiSet<T>, SortedMultiSet<T>, SortedMultiSet<T>>> Sum() => sum;
 
         public abstract Expression<Func<SortedMultiSet<T>, R>> ComputeResult();
+
+        private static SortedMultiSet<T> Apply(SortedMultiSet<T> state, Action<SortedMultiSet<T>> op)
```

Comment:
>Apply [](start = 41, length = 5)

What is the benefit of Apply here? it looks like the operations were already applied directly without an additional function call.

Reply:
removed this